mmtrack/models/sot/stark.py

```diff
@@ -83,6 +93,254 @@ def extract_feat(self, img):
+            search_area_factor (float): Ratio of crop size to target size
+            output_size (float): the size of output cropped image
+                (always square).
+        returns:
+            img_crop_padded (Tensor): of shape (1, C, output_size, output_size)
+            resize_factor (float): the ratio of original image scale to cropped
+                image scale.
+            pdding_mask (Tensor): the padding mask caused by cropping. It's
+                of shape (1, output_size, output_size).
+        """
+        cx, cy, w, h = target_bbox.split((1, 1, 1, 1), dim=-1)
+
+        img_h, img_w = img.shape[2:]
+        # 1. Crop image
+        # 1.1 calculate crop size and pad size
+        crop_size = math.ceil(math.sqrt(w * h) * search_area_factor)
+        if crop_size < 1:
+            raise Exception('Too small bounding box.')
+
+        x1 = torch.round(cx - crop_size * 0.5).long()
+        x2 = x1 + crop_size
+        y1 = torch.round(cy - crop_size * 0.5).long()
+        y2 = y1 + crop_size
+
+        x1_pad = max(0, -x1)
+        x2_pad = max(x2 - img_w + 1, 0)
+        y1_pad = max(0, -y1)
+        y2_pad = max(y2 - img_h + 1, 0)
+
+        # 1.2 crop image
+        img_crop = img[..., y1 + y1_pad:y2 - y2_pad, x1 + x1_pad:x2 - x2_pad]
+
+        # 1.3 pad image
+        img_crop_padded = F.pad(
+            img_crop,
+            pad=(x1_pad, x2_pad, y1_pad, y2_pad),
+            mode='constant',
+            value=0)
+        # 1.4 generate padding mask
+        _, _, img_h, img_w = img_crop_padded.shape
+        end_x = None if x2_pad == 0 else -x2_pad
+        end_y = None if y2_pad == 0 else -y2_pad
+        padding_mask = torch.ones((img_h, img_w),
+                                  dtype=torch.float32,
+                                  device=img.device)
+        padding_mask[y1_pad:end_y, x1_pad:end_x] = 0.
+
+        # 2. Resize cropped image and padding mask
+        resize_factor = output_size / crop_size
+        img_crop_padded = F.interpolate(
+            img_crop_padded, (output_size, output_size),
+            mode='bilinear',
+            align_corners=False)
+
+        padding_mask = F.interpolate(
+            padding_mask[None, None], (output_size, output_size),
+            mode='bilinear',
+            align_corners=False).squeeze(dim=0).type(torch.bool)
+
+        return img_crop_padded, resize_factor, padding_mask
+
+    def init(self, img, bbox):
+        """Initialize the single object tracker in the first frame.
+
+        Args:
+            img (Tensor): input image of shape (1, C, H, W).
+            bbox (list | Tensor): in [cx, cy, w, h] format.
+        """
+        self.z_dict_list = []  # store templates
+        # get the 1st template
+        z_patch, _, z_mask = self.get_cropped_img(
+            img, bbox, self.test_cfg['template_factor'],
+            self.test_cfg['template_size']
+        )  # z_patch of shape [1,C,H,W];  z_mask of shape [1,H,W]
+        z_patch = normalize(
```

Comment:
suggest use [mmcv.imnormalize](https://github.com/open-mmlab/mmcv/blob/5b5b47fc87feb5cbd80197fd8d2c81da7a44de1a/mmcv/image/photometric.py#L9)

Reply:
ignore this comment, just remember that `[mmcv.imnormalize](https://github.com/open-mmlab/mmcv/blob/5b5b47fc87feb5cbd80197fd8d2c81da7a44de1a/mmcv/image/photometric.py#L9)` needs ndarray.

Reply:
Fix the lint error, then we can merge this PR.